### DISCUSSION TITLE
Prerender and partial hydration

### DISCUSSION THREAD

**Discussion by akagusu** [preactjs/wmr#655] [Q&A] [answered]
I have pages that are almost 100% text. There are only a few interactive widgets.

Is it possible to prerender routes with client-side hydration disabled and then hydrate only interactive components?

I'm in a region where mobile connections have poor quality and phones are mostly entry-level phones with low resources. I know Preact is small and performant but here every kb counts.

Thanks.

**Comment by developit** [comment, marked as answer]
The short answer is that [this is possible](https://gist.github.com/developit/e94cd0da8479aacd1bbdedd612c1975f), but we don't yet provide an automated way to do it. However, it's super high on my wishlist/todo list to publish a plugin for this - there is a lot of performance potential that this unlocks if done right.

**Reply by akagusu** [reply]
Good to know you're thinking about this.
I read your article about islands architecture and that is exactly what I need. I know the guys from [Astro](https://astro.build/) are working in something like this. It would be really awesome to see a Preact implementation of this.
